packages/react/src/components/Toast/index.tsx

```diff
@@ -0,0 +1,113 @@
+import React, {
+  PropsWithChildren,
+  ReactNode,
+  useEffect,
+  useMemo,
+  useRef,
+} from "react";
+import classNames from "classnames";
+import { ToastType } from ":/components/Toast/ToastProvider";
+import { typeToColor } from ":/components/Alert/Utils";
+import { Button, ButtonProps } from ":/components/Button";
+
+export interface ToastProps extends PropsWithChildren {
+  duration: number;
+  type: ToastType;
+  onDelete?: () => void;
+  icon?: ReactNode;
+  primaryLabel?: string;
+  primaryOnClick?: ButtonProps["onClick"];
+  primaryProps?: ButtonProps;
+  disableAnimate?: boolean;
+  actions?: ReactNode;
+}
+
+export const Toast = (props: ToastProps) => {
+  const [animateDisappear, setAnimateDisappear] = React.useState(false);
+  const container = useRef<HTMLDivElement>(null);
+
+  // Register a timeout to remove the toast after the duration.
+  useEffect(() => {
+    if (props.disableAnimate) {
+      return;
+    }
+    setTimeout(async () => {
+      setAnimateDisappear(true);
+    }, props.duration);
+  }, []);
```

Comment:
You should save the timeout id into a ref then clear the timeout on component unmount.